aten/src/ATen/native/Pow.cpp

```diff
@@ -81,6 +81,60 @@ Tensor pow(Scalar base, const Tensor& exp) {
   return native::pow_out(result, base, exp);
 }
 
+Tensor& float_power_out(Tensor& result, const Tensor& base, const Tensor& exp) {
+  auto dtype = (at::isComplexType(base.scalar_type()) || at::isComplexType(exp.scalar_type())) ?
+                at::kComplexDouble : at::kDouble;
+  TORCH_CHECK(result.scalar_type() == dtype,
+              "output type ", result.scalar_type(), "is not the desired output type ", dtype);
+
+  return at::pow_out(result, base.to(dtype), exp.to(dtype));
+}
+
+Tensor& float_power_out(Tensor& result, const Tensor& base, Scalar exp) {
```

Comment:
Should this Tensor x Scalar variant and the next Scalar x Tensor variant both call the prior Tensor x Tensor variant after wrapping the scalar in a tensor? That might make the code more maintainable.

Reply:
Addressed! Thanks for advice!